task/task.js

```diff
@@ -0,0 +1,40 @@
+export class Task{
+    constructor(onUpdate, onDelete, onDone, onUnDone){
+        this.done = false; // this will mark if the task has been marked as done or not
+        this.text = "";
+
+        this.onUpdate = onUpdate;
+        this.onDelete = () => {onDelete(this)};
```

Comment:
do it with `bind`

Reply:
using bing will force me to change the "this" parameter of the function, which i do not want, i want the function to retain its original "this" attribute and be a function that gets and input and applies on it.
In the case that i pass this function to the task object, i want the task to always pass itself as the parameter to the function.